server/src/service-layer/controllers/UserSignup.ts

```diff
@@ -0,0 +1,37 @@
+import AuthService from "business-layer/services/AuthService"
+import { UserAdditionalInfo } from "data-layer/models/firebase"
+import UserDataService from "data-layer/services/UserDataService"
+import { UserSignupBody } from "service-layer/request-models/UserSignupRequests"
+import { Body, Controller, Post, Route, SuccessResponse } from "tsoa"
+
+@Route("signup")
+export class UserSignup extends Controller {
+  @Post()
+  @SuccessResponse(200, "Signup successful")
+  // return a JWT token at the end
+  public async signup(@Body() requestBody: UserSignupBody): Promise<string> {
+    const userService = new UserDataService()
+    const authService = new AuthService()
+
+    // Omit membership to avoid users from creating an account with admin
+    const userInfo: Omit<UserAdditionalInfo, "membership"> = requestBody.user
```

Comment:
we should omit this information in the request body type, so that it doesn't appear on the our API docs. (Also need to omit `stripe_id`